src/main/scala/vectorpipe/VectorPipe.scala

```diff
@@ -132,14 +129,19 @@ object VectorPipe {
     /* Associate each Feature with a SpatialKey */
     val grid: RDD[(SpatialKey, Feature[G, D])] = rdd.flatMap(f => byIntersect(mt, f))
 
+    /** A way to render some Geometry that failed to clip. */
+    val errorClipping: ((Extent, Feature[G, D])) => String = { case (e, f) =>
+      s"CLIP FAILURE W/ EXTENT: ${e}\nELEMENT METADATA: ${f.data}\nGEOM: ${f.geom.reproject(WebMercator, LatLng).toGeoJson}"
+    }
+
     grid.groupByKey().map { case (k, iter) =>
       val kExt: Extent = mt(k)
 
       /* Clip each geometry in some way */
       val clipped: List[Feature[G, D]] = iter.foldLeft(List.empty[Feature[G, D]]) { (acc, g) =>
         clip(kExt, g) match {
           case Some(h) => h :: acc
-          case None => logError(kExt, g); acc /* Sneaky IO to log the error */
+          case None => logError(errorClipping)((kExt, g)); acc /* Sneaky IO to log the error */
```

Comment:
Being sneaky w/r/t the type system is unavoidable, but I think you should break this out and put the `IO` and the return value on separate lines for human-sake

Reply:
Fair, fair.